docs/internal/GeneralArchitectureGuide.md

```diff
@@ -68,6 +68,50 @@ state must ever be reloaded from persisted state.
 
 ## Deprecations
 
+## Backwards Compatibility
+
+Elasticsearch versions are composed of three pieces of information: the major version, the minor version, and the patch version,
+in that order (major.minor.patch). Patch releases are typically bug fixes; minor releases are improvements; and major releases
+contain new features. As an example, each of 8.0.0, 8.3.0 and 8.3.1 specifies an exact release version. They all have the same
```

Comment:
Typically new features land in minor releases; major releases are mostly about breaking compatibility and dropping deprecated functionality.

Reply:
Thanks, it didn't seem quite right when I wrote it.